### PR TITLE
Drop support for Django 1.7

### PR DESCRIPTION
**Pull request by michael-k**
As suggested in https://github.com/pytest-dev/pytest-django/pull/521#discussion_r143324799

### CODE REVIEW

File: pytest_django/compat.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # This file cannot be imported from until Django sets up
+from django.db.backends.base.base import BaseDatabaseWrapper
```
Comment (michael-k) [inline pytest_django/compat.py]: Is this (`pytest_django.compat`) a public API?  If not, we could remove it and adjust the import in `plugin.py`.

**Comment by blueyed** [comment]
Thanks for tackling this - the diff should be covered 100% though in this case I think.

**Comment by michael-k** [comment]
> the diff should be covered 100% though in this case I think.

I'll look into it this weekend or so.

**Comment by blueyed** [comment]
Looking forward to it.
Let's wait for https://github.com/pytest-dev/pytest-django/pull/521 to be merged first.

**Comment by blueyed** [comment]
#521 is merged, so please rebase.

**Comment by michael-k** [comment]
> please rebase

done

**Comment by blueyed** [comment]
close-reopen again for codecov to report back hopefully.

**Comment by blueyed** [comment]
Do you feel like covering the diff for this?
https://codecov.io/gh/pytest-dev/pytest-django/compare/16097801330500399d210d423689d71e18514c78...1a8bfe20feaab4d7e5f6553c035ce0262df7fd0b/diff

**Comment by michael-k** [comment]
Give me a few hours to look over it.  I'll report back as soon as I have an idea of how much effort this requires.

**Comment by michael-k** [comment]
- .
    ```
    # fixtures.py
    def django_db_setup(…):
        …
        if django_db_keepdb:
            if get_django_version() >= (1, 8):  # removed in this PR
                setup_databases_args['keepdb'] = True
            else:  # removed in this PR
                …  # removed in this PR
        …
    ```

    `django_db_keepdb` is `True` if `request.config.getvalue('reuse_db') and …`, but `reuse_db`/`reuse-db` is only set in `test_db_setup.py`. I do not know why this does not cover the above code.

- The other thing that changed without coverage ist `if get_django_version() >= (1, 8) and dj_settings.TEMPLATES:` to `if dj_settings.TEMPLATES:` (2x) in `plugin.py`.  `def _fail_for_invalid_template_variable(request)` is not covered at all and `def _template_string_if_invalid_marker(request)` has only one fully covered line.  I have no idea how to cover the missing lines.

*Conclusion:*
The lines were not tested before.  The changes (effectively replacing `get_django_version() >= (1, 8)` with `True` in conditions) don't affect the codebase at all for Django 1.8+. Diff coverage is missing because tests are missing, not because this PR made anything worse.
I do not regard the missing tests as an “easy pick”, so I'd say that they are out of scope of this PR. :(

**Comment by blueyed** [comment]
Thanks!

**Pull request merged**